### ISSUE TITLE
AttributeError: can't set attribute

### ISSUE THREAD

**Issue by lukik** [Corvia/django-tenant-users#20]
Created a fresh Django project to understand `djang-tenant-users`. The project has only two apps `tenants` and `authy` for the custom user. 

After following the guide on how to use django-tenant-users, am getting the error `AttributeError: can't set attribute` when I run `makemigrations` or `migrate_schemas`. 

```
  File "manage.py", line 39, in <module>
    main()
  File "manage.py", line 35, in main
    execute_from_command_line(sys.argv)
  File "/Volumes/storage/apps/venv/tenant_env/lib/python3.6/site-packages/django/core/management/__init__.py", line 371, in execute_from_command_line
    utility.execute()
  File "/Volumes/storage/apps/venv/tenant_env/lib/python3.6/site-packages/django/core/management/__init__.py", line 365, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Volumes/storage/apps/venv/tenant_env/lib/python3.6/site-packages/django/core/management/base.py", line 288, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Volumes/storage/apps/venv/tenant_env/lib/python3.6/site-packages/django/core/management/base.py", line 332, in execute
    self.check()
  File "/Volumes/storage/apps/venv/tenant_env/lib/python3.6/site-packages/django/core/management/base.py", line 364, in check
    include_deployment_checks=include_deployment_checks,
  File "/Volumes/storage/apps/venv/tenant_env/lib/python3.6/site-packages/django/core/management/base.py", line 351, in _run_checks
    return checks.run_checks(**kwargs)
  File "/Volumes/storage/apps/venv/tenant_env/lib/python3.6/site-packages/django/core/checks/registry.py", line 73, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/Volumes/storage/apps/venv/tenant_env/lib/python3.6/site-packages/django/contrib/auth/checks.py", line 74, in check_user_model
    if isinstance(cls().is_anonymous, MethodType):
  File "/Volumes/storage/apps/venv/tenant_env/lib/python3.6/site-packages/django/db/models/base.py", line 480, in __init__
    _setattr(self, field.attname, val)
AttributeError: can't set attribute
```

Apps Layout
```
SHARED_APPS = (
    'django.contrib.sessions',
    'django.contrib.sites',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.humanize',  # Useful templates tags:

    'django_tenants',  # tomturner/django-tenants
    'tenant_users.permissions',
    'tenant_users.tenants',

    'tenants',
    'authy', 
)

TENANT_APPS = (
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'tenant_users.permissions',
)

INSTALLED_APPS = list(SHARED_APPS) + [app for app in TENANT_APPS if app not in SHARED_APPS]
```

Packages in Use
> Django==2.0.6
> django-tenants==2.0.0
> django-tenant-users==0.3.11

_FYI, if I remove the `django-tenant-users` settings and remain only with `django-tenants` settings, I don't get any error._

**Update**
I've narrowed down the issue to a field in my custom user model.

`is_staff = models.BooleanField(default=False)`

When that specific field is present, the above error occurs. When its not, `makemigrations` and other commands execute properly. Why would that be?

**Comment by ivissani** [comment]
I think that the problema may be because your defining a field that clashes with the following property:

```python
@property
    def is_staff(self):
        try:
            _is_staff = self._get_tenant_perms().is_staff
            return _is_staff
        except UserTenantPermissions.DoesNotExist:
            return False
```
in the `PermissionsMixinFacade` class. Note that `UserProfile` in `django-tenant-users` already implements `is_staff` property in a per-tenant basis.

**Comment by lukik** [comment]
Thanks. I sure do wish there was a better error message when one encounters it...

**Comment by lukik** [comment]
This has came to bite me again. 

I installed [django-multisite](https://github.com/ecometrica/django-multisite) so as to be able to activate admin. I tried logging in via Django Admin but failed. 

If you check the tenant models created by `django-tenant-users`, you'll find the `permissions_usertenantpermissions` model. For the initial user created via console, it has the property of `is_superuser=True` and `is_staff=False`. 

If I change `is_staff` to `True`, am able to login via Django Admin portal. 

I would have expected `is_staff` to be True by default when you create a Superuser? Or what am I missing?
